blog/content/post/2021-07-13-ftpm-vs-dtpm.md

```diff
@@ -0,0 +1,125 @@
+---
+title: fTPM vs dTPM
+abstract: "An introduction to TPMs.
+           Let's explore the differences between common implementations of TPMs
+           and how they might matter to you."
+cover: /covers/tpm2.png
+author:
+    - michal.kopec
+layout: post
+published: true
+date: 2021-07-13
+archives: "2021"
+
+tags:
+  - tpm
+  - firmware
+  - hardware
+categories:
+  - Security
+
+---
+
+## Intro
+
+One of the most important topics in firmware security is TPM. You may have just
+heard about them in news recently as Microsoft started requiring a TPM 2.0 for
+Windows 11. This has been a source of great confusion as to why it's so
+important, some have even theorized it's for DRM and restricting user control.
+In this article we'll learn what TPMs are and we'll explore the differences
+between the most common implementations: fTPM and dTPM.
+
+## What's a TPM?
+
+TPMs aren't actually all that new. The first implementation of a dedicated chip
+resembling a TPM debuted in the IBM ThinkPad T23 in 2001, and implemented
+features like password storage and asymmetric cryptography functions. It wasn't
+much more functional than a smartcard.
+
+In 2003, IBM joined up with various companies including HP, Intel and Microsoft
+and established the Trusted Computing Platform Alliance (TCPA), later known as
+the the Trusted Computing Group (TCG). The TCPA defined the TPM standard, which
+has continuously evolved over the years, picking up adoption especially in
+enterprise and business use.
+
+The TPM's primary functions are to store secrets in isolation from the host OS
+and verify system integrity by storing measurements of the boot proces in PCR
+registers. Without going into too much detail, it can verify the integrity of
+every component in the boot process and allows the establishment of a trusted
+computing platform.
+
+While TPMs began as dedicated security chips, as they picked up adoption,
+silicon vendors started implementing them in chipsets and eventually in firmware.
+A firmware based implementation is commonly called a Firmware TPM, or fTPM for
+short, runs in the CPU's trusted execution environment, doesn't take up any
+extra space on the board which is great for space-constrained boards, and is
+drastically cheaper to implement than a dedicated chip. A hardware based
+implementation in a discrete package is now called Discrete TPM, or dTPM.
+
+You may have also heard about sTPM, a software implementation of TPM running in
+userspace, but as it's not at all isolated from the host OS (without the use of
+technologies like Intel SGX, at least) it doesn't offer any real security
+benefits. It's mostly just useful for development purposes.
```

Comment:
I would be good to mention sTPM usability in hypervisor environment where it can be used as a vTPM for example.

Reply:
Quite a lot of presentations discuss that:
https://youtu.be/Eip5Rts6S2I
https://youtu.be/pZF-jyJWTE4
https://youtu.be/y3V_V0Vllas?t=11447